### PR TITLE
whisper-ctranslate2: init at 0.2.7

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

addition to the existing packages.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 240833` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.whisper-ctranslate2</li>
    <li>python310Packages.whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
This looks more like an application than a library, should it be under python310Packages?

**Comment by happysalada** [comment]
you're right this doesn't export a python module at all.
I've moved it to the tools audio section instead.

File: pkgs/tools/audio/whisper-ctranslate2/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, nix-update-script
+, python3
+, python3Packages
```
Comment (natsukium) [inline pkgs/tools/audio/whisper-ctranslate2/default.nix:5]:
```suggestion
```

File: pkgs/tools/audio/whisper-ctranslate2/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    hash = "sha256-dUmQNKgH+SIlLhUEiaEGXUHZQDr3fidsAU2vATJiXBU=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    numpy
+    faster-whisper
+    ctranslate2
+    tqdm
+    sounddevice
+    tqdm
```
Comment (natsukium) [inline pkgs/tools/audio/whisper-ctranslate2/default.nix:30]:
```suggestion
```

File: pkgs/tools/audio/whisper-ctranslate2/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description = "Whisper command line client compatible with original OpenAI client based on CTranslate2";
+    homepage = "https://github.com/Softcatala/whisper-ctranslate2";
+    changelog = "https://github.com/Softcatala/whisper-ctranslate2/releases/tag/${version}";
+    license = licenses.asl20;
```
Comment (natsukium) [inline pkgs/tools/audio/whisper-ctranslate2/default.nix:39]:
```suggestion
    license = licenses.mit;
```
It looks like MIT License.
https://github.com/Softcatala/whisper-ctranslate2/blob/f24872f7885451aba982c5d42616aa0386e1884a/LICENSE#L1

File: pkgs/tools/audio/whisper-ctranslate2/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+, fetchFromGitHub
+, nix-update-script
+, python3
+}:
+let
+  pname = "whisper-ctranslate2";
+  version = "0.2.7";
+in
+python3.pkgs.buildPythonApplication {
+  inherit pname version;
```
Comment (natsukium) [inline pkgs/tools/audio/whisper-ctranslate2/default.nix:11]:
```suggestion

python3.pkgs.buildPythonApplication rec {
  pname = "whisper-ctranslate2";
  version = "0.2.7";
```
I often see this pointed out, but I don't think it needs to be changed necessarily.

**Review by natsukium — APPROVED** [review]

File: pkgs/tools/audio/whisper-ctranslate2/default.nix (hunk 4 above)
Comment (happysalada) [reply pkgs/tools/audio/whisper-ctranslate2/default.nix:11]: Thanks for taking the time to mention it.
Im actually against rec, some people mention it as an anti pattern, some people love it. What i personally dont like is that you cant tell at a glance what is being used where. Using let in is slightly more verbose, but personally i find it clearer.

Comment (natsukium) [reply pkgs/tools/audio/whisper-ctranslate2/default.nix:11]: Thanks for telling me why.
> What i personally dont like is that you cant tell at a glance what is being used where.

I agree with this.

**Pull request merged**